featurebyte/query_graph/sql/interpreter/tile.py

```diff
@@ -140,17 +140,31 @@ def _get_event_table_timestamp_filter(
             groupby_input_node
         )
         op_struct = op_struct_info.operation_structure_map[groupby_input_node.name]
+
         for column in op_struct.columns:
-            if (
-                column.name == groupby_node.parameters.timestamp
-                and isinstance(column, SourceDataColumn)
-                and column.table_id is not None
-            ):
-                assert isinstance(column, SourceDataColumn)
-                return EventTableTimestampFilter(
-                    timestamp_column_name=column.name,
-                    event_table_id=column.table_id,
-                )
+            if column.name != groupby_node.parameters.timestamp:
+                continue
+
+            # Check if the timestamp column is from an EventTable
+            if isinstance(column, SourceDataColumn):
+                if column.table_type == TableDataType.EVENT_TABLE and column.table_id is not None:
+                    return EventTableTimestampFilter(
+                        timestamp_column_name=column.name,
+                        event_table_id=column.table_id,
+                    )
+            elif isinstance(column, DerivedDataColumn) and set(column.transforms) == {"join"}:
+                # In ItemView, the event timestamp column is considered a derived column because
+                # it is derived from join.
```

Comment:
Any issue if we drop ` and set(column.transforms) == {"join"}`?
The reason that I am asking this question is that `column.transforms` attribute currently is not really used in any critical operation (only used in deriving feature info). Would be good if we could avoid using this attribute.

Reply:
🤔 I think i know why it is used here.

Other option is doing the following:
1. Extract the node from the graph based on `column.node_name`.
2. If the node parameter is from the `ItemViewGraphNodeParameters`, we are safe to apply the following filter.